datafusion/functions/src/unicode/strpos.rs

```diff
@@ -40,8 +41,20 @@ impl Default for StrposFunc {
 
 impl StrposFunc {
     pub fn new() -> Self {
+        use DataType::*;
         Self {
-            signature: Signature::user_defined(Volatility::Immutable),
+            signature: Signature::one_of(
+                vec![
+                    Exact(vec![Utf8, Utf8]),
+                    Exact(vec![Utf8, LargeUtf8]),
+                    Exact(vec![LargeUtf8, Utf8]),
+                    Exact(vec![LargeUtf8, LargeUtf8]),
+                    Exact(vec![Utf8View, Utf8View]),
+                    Exact(vec![Utf8View, Utf8]),
+                    Exact(vec![Utf8View, LargeUtf8]),
+                ],
+                Volatility::Immutable,
```

Comment:
This restores the code before https://github.com/apache/datafusion/commit/1b3608da7ca59d8d987804834d004e8b3e349d18 . These are the types actually supported by invoke.

Reply:
Can we invert the order? From another function we have this comment:

// Planner attempts coercion to the target type starting with the most preferred candidate.
// For example, given input `(Utf8View, Int64)`, it first tries coercing to `(Utf8View, Int64)`.
// If that fails, it proceeds to `(Utf8, Int64)`.

Reply:
Good comment. I just restored the code that used to be here.
How function implementor can know what is the preferred order? Shouldn't this rather be engine's responsibility, if its given a choice? Unless of course the preferred candidate is function-specific.

Reply:
Great question and I don't have the answer. I just have followed that advice since I first saw it a month or two back. I agree that I would hope the coercing logic could select the most optimal candidate.